File: cakk-domain/src/main/java/com/cakk/domain/repository/jpa/UserJpaRepository.java
package com.cakk.domain.repository.jpa;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.cakk.domain.entity.user.User;

public interface UserJpaRepository extends JpaRepository<User, Long> {

	Optional<User> findByProviderId(String providerId);
}

File: cakk-infrastructure/persistence-mysql/src/main/java/com/cakk/infrastructure/persistence/repository/jpa/UserJpaRepository.kt
package com.cakk.infrastructure.persistence.repository.jpa

import com.cakk.infrastructure.persistence.entity.user.UserEntity
import org.springframework.data.jpa.repository.JpaRepository

interface UserJpaRepository : JpaRepository<UserEntity, Long> {

    fun findByProviderId(providerId: String): UserEntity?
}

